### PR TITLE
ADOBE-222: added new option to file download

### PR DESCRIPTION
**Pull request by ChernovDmytro**

### CODE REVIEW

File: api/files/index.ts [hunk 1]
```diff
@@ -41,13 +41,17 @@ export class SmartlingFilesApi extends SmartlingBaseApi {
     }
 
     async downloadFile(
-        projectId: string, fileUri: string, locale: string, params: DownloadFileParameters
+        projectId: string,
+        fileUri: string,
+        locale: string,
+        responseType: ResponseBodyType | boolean,
```
Comment (dimitrystd) [inline api/files/index.ts:47]: It looks weird. Why not add the 3rd enum value: `PARSE`, `TEXT`, `ARRAY_BUFFER`.
Well, no need to fix here but looks strange.

File: test/files.spec.ts [hunk 2]
```diff
@@ -112,7 +112,7 @@ describe("SmartlingFilesApi class tests.", () => {
                     .setRetrievalType(RetrievalType.PUBLISHED)
                     .enableDebugMode();
 
-                await filesApi.downloadFile(projectId, fileUri, localeId, params);
+                await filesApi.downloadFile(projectId, fileUri, localeId, true, params);
```
Comment (dimitrystd) [inline test/files.spec.ts:115]: Maybe pass `TEXT` here. It's not important for this test, but at least no question about what does `true` mean for `responseType`? Is it important for the test?

**Review by dimitrystd — APPROVED** [review]

File: api/files/index.ts (hunk 1 above)
Comment (PavelLoparev) [reply api/files/index.ts:47]: [It's already there](https://github.com/Smartling/api-sdk-nodejs/blob/master/api/base/enum/response-body-type.ts#L2). I think @ChernovDmytro was confused by [this line](https://github.com/Smartling/api-sdk-nodejs/blob/master/api/base/index.ts#L93) in the API SDK base class, but this was only for "backward compatibility" in all the children classes (to not to update them all). [In reality](https://github.com/Smartling/api-sdk-nodejs/blob/master/api/base/index.ts#L138) `ResponseBodyType.TEXT` is the same as using `true`, please remove `| boolean` typing from signature.

**Review by PavelLoparev — CHANGES_REQUESTED** [review]
I don't get the reason for this PR at all. it works the same after this PR as it was before the change

File: api/files/index.ts (hunk 1 above)
Comment (PavelLoparev) [reply api/files/index.ts:47]: Agree. It comes from https://github.com/Smartling/api-sdk-nodejs/blob/master/api/base/index.ts#L93 - I only done that to not to make changes in children classes.

**Review by PavelLoparev — CHANGES_REQUESTED** [review]
Oh, it only worked with text files before and doesn't work with binary indd, got it.

@ChernovDmytro please remove ` | boolean` from the method signature, `true` [is the same](https://github.com/Smartling/api-sdk-nodejs/blob/master/api/base/index.ts#L138) as passing `ResponseBodyType.TEXT` (also use `ResponseBodyType.TEXT` in tests as well).

File: test/files.spec.ts (hunk 2 above)
Comment (ChernovDmytro) [reply test/files.spec.ts:115]: I looked at implementation of FileDownload function of FIle Api and there was passed true, so I assumed this is correct way. But I agree taht better to add third option which will handle other cases.

**Review by PavelLoparev — CHANGES_REQUESTED** [review]
Wait, please add `responseType` as a last param with default value for backward compatibility.

**Review by PavelLoparev — APPROVED** [review]

**Pull request merged**